snap/snapcraft.yaml

```diff
@@ -33,7 +33,7 @@ grade: stable
 
 base: core18
 confinement: strict
-assumes: [snapd2.40]
+assumes: [snapd2.48.3]
```

Comment:
Soooo, this is complicated.  We need the latest `snapd` (which includes https://github.com/snapcore/snapd/pull/9764, necessary to make this bump even work), which in all the distro builds (https://packages.ubuntu.com/search?keywords=snapd) _is_ 2.48.3 but in https://snapcraft.io/snapd is 2.48.2.1. :grimacing:

I guess it's *probably* rare that someone is going to be updating `docker` but _not_ updating `snapd`, so we're probably safe to just go with 2.48.2 for now?

Reply:
for the purposes of that pr, then yes `assumes: [snapd2.48.2]` is sufficient, snapd2.48.2.1 would be _best_, but that is currently broken (but will eventually be fixed by https://github.com/snapcore/snapd/pull/9923)

Reply:
oh wait sorry I forgot that pr didn't make it into 2.48.2, gosh this is all so confusing to keep track of

Reply:
well I can say for sure for sure that snapd2.49 will have that pr in it, 2.49 is in beta now and should move to candidate/stable within the next 2 weeks (2.49 was blocked by the CVE)

Reply:
and checking in on the upgrade situation, we still have a sizable ( more than 10% of total userbase ) install base of 2.48.2 on core/snapd snaps (across all devices, not necessarily docker snap users), so if we do include 2.48.2 in the assumes it's possible that folks are still on 2.48.2 and as such they won't be _required_ to update snapd before installing docker. 

at this point I'm feeling like since we pushed out something that triggered refreshes to deal with that CVE, we should just make this wait until 2.49 :-|

Reply:
Heh, nice!  I'm honestly a little confused why all the suites of Ubuntu got 2.48.3 but the store got 2.48.2.1 which appears to be the same underlying version?  I would've expected that to be the other way around (if anything), TBH.

I think the likelihood of folks updating `docker` but not having updated `snapd` is terribly low, but it would be nice to have that actually validated (without having to wait another week or two to get there).  I guess we could just plan to tell anyone who complains to update their `snapd`?  I don't get to see https://snapcraft.io/snapd/metrics, but I _have_ to imagine the number of (data reporting) users who didn't get the bump by now is pretty tiny.  I'd worry more about users who aren't consuming `snapd` from the Snap, but their mitigation path is pretty straightforward. :sweat_smile:

Reply:
In all likelihood, I think you're correct in that the number of affected docker snap users will likely be very small, so we are probably good to just go with `assumes: [snapd2.48.2]`, and then after 2.49 is out maybe we can revisit and bump it again to 2.49 to be super extra sure